src/cmd-build-tag

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3 -u
+# Allows users to operate on the tags in `builds.json`
+
+import argparse
+import json
+import os
+import re
+import sys
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import fatal, rfc3339_time, write_json
+
+
+supported_actions = ["create", "delete", "list-builds", "list-tags", "update"]
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--force", action="store_true",
+                    help="Force creating/updating of tag")
+parser.add_argument("action", choices=supported_actions, nargs=1,
+                    help=f"Action to perform")
+parser.add_argument("tag", nargs="?",
+                    help="Tag to operate on")
+parser.add_argument("build", nargs="?",
+                    help="Build to associate tag with")
+args = parser.parse_args()
+
+# Read existing builds.json
+builds_json = "builds/builds.json"
+if os.path.isfile(builds_json):
+    with open(builds_json) as f:
+        builddata = json.load(f)
+else:
+    builddata = {"builds": []}
+
+# Validate arguments
+if (not args.force and
+        (args.action[0] == "create" or args.action[0] == "update")):
+    if args.tag is None:
+        fatal(f"Must supply tag when performing {args.action[0]}")
+
+    if args.build is None:
+        fatal(f"Must supply build when performing {args.action[0]}")
```

Comment:
With the subparser approach, things like this could be enforced by `argparse` itself.